pyproject.toml

```diff
@@ -39,6 +39,7 @@ dependencies = [
   "build",
   "click >= 8",
   "pip >= 22.2",
+  "pyproject_hooks",
```

Comment:
I tried but it does not work with `pipprevious`

Reply:
We can pin `pip>=23.0` in `pyproject.toml` (in a separate PR) so that we could use [vendored](https://github.com/pypa/pip/commit/fa4b2efbabffb96536c10bfb94477912a0f20e7f) version.

Reply:
If you open a separate PR I'll gladly help with the review.

Reply:
Like [so](https://github.com/jazzband/pip-tools/pull/1971)?

Reply:
> We can use vendored package `pip._vendor.pyproject_hooks` instead.

@atugushev I think it doesn't make much sense to special-case this since we already depend on `build` unconditionally and `pyproject_hooks` is pulled in as a transitive dependency anyway.

Reply:
Makes sense. Thanks.